En_Bji_01 (Shikashi) OK and partially documented.

rylieb opened the pull request:
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---
<!-- Leave the text above intact. Add additional comments below. -->
![En_Bji_01 gv](https://user-images.githubusercontent.com/31797144/126424680-f4ccadff-1be4-4e7f-a080-e5e429ee0a67.png)

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.h:
```diff
@@ -2,12 +2,31 @@
+    /* 0x2A6 */ s16 headZRotAdj;
+    /* 0x2A8 */ s16 headXRotAdj;
+    /* 0x2AA */ u16 textId;
+    /* 0x2AC */ s16 cutscenes[1]; /* s16 alignment pad by compiler */
```
There's no need to add this comment.
```suggestion
    /* 0x2AC */ s16 cutscenes[1];
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.h:
```diff
@@ -2,12 +2,31 @@
 #define Z_EN_BJI_01_H
 
 #include <global.h>
+#include "../ovl_Obj_Moon_Stone/z_obj_moon_stone.h"
```
We don't use relative includes.
```suggestion
#include "overlays/actors/ovl_Obj_Moon_Stone/z_obj_moon_stone.h"
```

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -9,7 +9,33 @@ void EnBji01_Destroy(Actor* thisx, GlobalContext* globalCtx);
+s32 EnBji01_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* actor);
+void EnBji01_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* actor);
+
+/* Segmented data */
```
The comments labelling helper functions and data are unneeded

rylieb replied:
Will be fixed in a batch commit.

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+void func_809CCDE0(EnBji01* this, GlobalContext* globalCtx) {
+
+    Player* player = PLAYER;
+
```
Extra line (I think format.sh removes these so maybe run that again, if it doesn't ignore me)

EllipticEllipsis replied:
Apparently it doesn't.

rylieb replied:
Will be fixed in a batch commit.

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+
+
+void func_809CCDE0(EnBji01* this, GlobalContext* globalCtx) {
+
```
Extra empty line here

rylieb replied:
Will be fixed in a batch commit.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+};
+
+/* Animations struct */
+static struct_80B8E1A8 D_809CDC7C[4] = {
```
There's no need to put the length of the array explicitly. We usually don't do this.
```suggestion
static struct_80B8E1A8 D_809CDC7C[] = {
```

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+}
+
+void func_809CCE98(EnBji01* this, GlobalContext* globalCtx) {
+
```
extra line

rylieb replied:
Will be fixed in a batch commit.

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+}
+
+void func_809CCEE8(EnBji01* this, GlobalContext* globalCtx) {
+
```
extra line

rylieb replied:
Will be fixed in a batch commit.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+};
+
+
+void func_809CCDE0(EnBji01* this, GlobalContext* globalCtx) {
+
+    Player* player = PLAYER;
+
+    Vec3f sp58;
+    s32 pad[2];
+
+    Math_Vec3f_Copy(&sp58, &player->actor.world.pos);
```
We keep the stack declaration together and with one line of separation from the rest of the code.
```suggestion
void func_809CCDE0(EnBji01* this, GlobalContext* globalCtx) {
    Player* player = PLAYER;
    Vec3f sp58;
    s32 pad[2];

    Math_Vec3f_Copy(&sp58, &player->actor.world.pos);
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                  &sp58, &this->headZRotAdj, &this->headXRotAdj, &this->torsoZRotAdj, &this->torsoXRotAdj, 0x1554, 0x1FFE, 0xE38, 0x1C70);
+}
+
+void func_809CCE98(EnBji01* this, GlobalContext* globalCtx) {
+
+    func_8013E1C8(&this->skelAnime, D_809CDC7C, 0, &this->animationIndex);
```
We don't leave a blank line like that. This also applies to the rest of the PR.
```suggestion
void func_809CCE98(EnBji01* this, GlobalContext* globalCtx) {
    func_8013E1C8(&this->skelAnime, D_809CDC7C, 0, &this->animationIndex);
```

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+        if ((this->actor.xzDistToPlayer <= 60.0f) && (this->actor.yDistToPlayer <= 10.0f)) {
+            this->actor.flags |= 0x10000;
+        } else {
+            this->actor.flags &= 0xFFFEFFFF;
```
I think this should be `this->actor.flags &= ~0x10000;`

rylieb replied:
I think I fixed this one earlier as well.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+        if ((this->actor.xzDistToPlayer <= 60.0f) && (this->actor.yDistToPlayer <= 10.0f)) {
+            this->actor.flags |= 0x10000;
+        } else {
+            this->actor.flags &= 0xFFFEFFFF;
```
This is a nicer syntax for negating that flag.
```suggestion
            this->actor.flags &= ~0x10000;
```

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+            this->actor.flags &= 0xFFFEFFFF;
+        }
+    }
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
```
`func_800B84D0` returns a bool so no need to check against 0 explicitly here

rylieb replied:
This will be fixed in an upcoming batch commit.

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+        globalCtx->msgCtx.unk11F22 = 0;
+        globalCtx->msgCtx.unk11F10 = 0;
+        func_809CD028(this, globalCtx);
+    } else {
```
```    
} else if (this->moonsTear != NULL && this->moonsTear->actor.colChkInfo.health == 1) {
                func_809CD6C0(this, globalCtx);
            }
} else {
            this->moonsTear =
                (ObjMoonStone*)func_ActorCategoryIterateById(globalCtx, NULL, ACTORCAT_PROP, ACTOR_OBJ_MOON_STONE);
        }
        func_800B8500(&this->actor, globalCtx, 60.0f, 10.0f, 0);
    }
}
```

kyleburnette replied:
This should probably match and is more clear + gets rid of the return

rylieb replied:
I tried it, but it actually doesn't match.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+        case 1: /* Have finished a conversation with Shikashi */
+            switch (gSaveContext.playerForm) {
+                case PLAYER_FORM_DEKU:
+                    if ((gSaveContext.weekEventReg[17] & 0x10) != 0) {
+                        if ((gSaveContext.weekEventReg[74] & 0x80) != 0) {
+                            this->textId = 0x5F4;
```
There's no need to leave the extra `!= 0` for those bitwise checks.
```suggestion
                    if (gSaveContext.weekEventReg[17] & 0x10) {
                        if (gSaveContext.weekEventReg[74] & 0x80) {
                            this->textId = 0x5F4;
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                case PLAYER_FORM_HUMAN:
+                    if (Player_GetMask(globalCtx) == PLAYER_MASK_KAFEIS_MASK) {
+                        this->textId = 0x236A;
+                    } else if ((gSaveContext.weekEventReg[74] & 0x10) != 0) {
```
```suggestion
                    } else if (gSaveContext.weekEventReg[74] & 0x10) {
```

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+}
+
+void func_809CD028(EnBji01* this, GlobalContext* globalCtx) {
+
```
extra line

rylieb replied:
This will be fixed in an upcoming batch commit.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                    break;
+                case PLAYER_FORM_GORON:
+                case PLAYER_FORM_ZORA:
+                    if ((gSaveContext.weekEventReg[75] & 8) != 0) {
```
```suggestion
                    if (gSaveContext.weekEventReg[75] & 8) {
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+        case 3: /* Currently engaged in post-telescope Shikashi dialogue */
+            switch (gSaveContext.playerForm) {
+                case PLAYER_FORM_DEKU:
+                    if ((gSaveContext.weekEventReg[74] & 0x80) != 0) {
```
```suggestion
                    if (gSaveContext.weekEventReg[74] & 0x80) {
```

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+    f32 tempTimeBeforeMoonCrash;
+
+    switch (this->actor.params) {
+        case 0: /* Have not talked to Shikashi as of entering observatory last */
```
I would create an enum in the header with the possible cases for this switch and move the comments there

EllipticEllipsis replied:
I wouldn't do it in the header since it's only used by this function, but an enum is a good idea.

rylieb replied:
This will be fixed in an upcoming batch commit.

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+}
+
+void func_809CD328(EnBji01* this, GlobalContext* globalCtx) {
+
```
extra line

rylieb replied:
This will be fixed in an upcoming batch commit.

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+            break;
+        case 4:
+            if (func_80147624(globalCtx) != 0) {
+                this->actor.flags &= 0xFFFEFFFF;
```
Same negated value as earlier in the review (~0x10000)

rylieb replied:
I think i already fixed this one earlier.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                        case 2:
+                            this->textId = 0x5EA;
+                            break;
+                        case 3:
+                            tempDay = gSaveContext.day;
+                            tempTimeBeforeMoonCrash =
+                                ((-(tempDay % 5 << 0x10) - ((u16)(gSaveContext.time - 0x4000))) + 0x40000);
+                            if (tempTimeBeforeMoonCrash < 2730.6667f /* 1 hr */) {
```
wtf?
Errr... well, if it matches...
Hopefully we could rewrite this in a more readable way eventually. There's no need for you to do it.

EllipticEllipsis replied:
```suggestion
                        case 3:
                            tempDay = gSaveContext.day;
                            tempTimeBeforeMoonCrash =
                                ((-(tempDay % 5 << 0x10) - ((u16)(gSaveContext.time - 0x4000))) + 0x40000);
                            if (tempTimeBeforeMoonCrash < 2730.6667f) { /* 1 hr */)
```
Would rather the comment was outside.

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                        func_8019F230();
+                        switch (gSaveContext.playerForm) {
+                            case PLAYER_FORM_DEKU:
+                                func_80151938(globalCtx, 0x5F0U);
```
For unsigned values like this you can remove the U

rylieb replied:
Will be fixed in a batch commit.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+            break;
+        case 4:
+            if (func_80147624(globalCtx) != 0) {
+                this->actor.flags &= 0xFFFEFFFF;
```
```suggestion
                this->actor.flags &= ~0x10000;
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                    case 1:
+                        func_8019F230();
+                        switch (gSaveContext.playerForm) {
+                            case PLAYER_FORM_DEKU:
+                                func_80151938(globalCtx, 0x5F0U);
+                                break;
+                            case PLAYER_FORM_HUMAN:
+                                func_80151938(globalCtx, 0x5F8U);
+                                break;
+                            case PLAYER_FORM_GORON:
+                            case PLAYER_FORM_ZORA:
+                                func_80151938(globalCtx, 0x5E1U);
+                                break;
```
There's no need for those extra `U` at the end.
```suggestion
                            case PLAYER_FORM_DEKU:
                                func_80151938(globalCtx, 0x5F0);
                                break;
                            case PLAYER_FORM_HUMAN:
                                func_80151938(globalCtx, 0x5F8);
                                break;
                            case PLAYER_FORM_GORON:
                            case PLAYER_FORM_ZORA:
                                func_80151938(globalCtx, 0x5E1);
                                break;
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+            break;
+        case 5:
+            if (func_80147624(globalCtx) != 0) {
+                this->actor.flags &= 0xFFFEFFFF;
```
```suggestion
                this->actor.flags &= ~0x10000;
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                    case 1518:
+                    case 1522:
+                    case 1525:
+                        func_80151938(globalCtx, (globalCtx->msgCtx.unk11F04 + 1) & 0xFFFF);
```
Would this match?
```suggestion
                        func_80151938(globalCtx, (u16)(globalCtx->msgCtx.unk11F04 + 1));
```

EllipticEllipsis replied:
Or maybe the prototype of this function is wrong. I assume something is needed?

rylieb replied:
& with 0xFFFF does match. In fact it's what the raw mips_to_c produced, but casting to u16 also matched, and I thought it was clearer, because that function takes a u16 as an argument, and &ing with 0xFFFF could be either s16 or u16.

rylieb replied:
actually, I thought this was the hours to moon crash calculation which looks very similar. I have never looked deeply at this function so I'm not sure.

AngheloAlf replied:
`& 0xFFFF` is not equivalent to an `(s16)` cast. IDO makes shifts operations to cast to signed types.

EllipticEllipsis replied:
`func_80151938` takes a u16 anyway, so it should match without.

rylieb replied:
I assume this will match without either u16 or &0xFFFF but I'm rebuilding so that I can run diff.py before I commit.

rylieb replied:
It does match without either, so I'll be pushing that change manually.

rylieb replied:
Will be fixed in a batch commit.

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CD70C.asm")
+    return 0;
```
`return false`

rylieb replied:
Will be fixed in a batch commit.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                        break;
+                    case 1520:
+                    case 1526:
+                        func_80151938(globalCtx, 0x5EFU);
```
```suggestion
                        func_80151938(globalCtx, 0x5EF);
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+                    case 1527:
+                    case 1528:
+                        func_801477B4(globalCtx);
+                        this->actor.flags &= 0xFFFEFFFF;
```
```suggestion
                        this->actor.flags &= ~0x10000;;
```

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+    s32 temp_f4 = 0;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/EnBji01_Update.asm")
+    if (limbIndex == 0xF) {
```
` if (limbIndex == 15)`

rylieb replied:
Will be fixed in a batch commit.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+            break;
+        case 6:
+            this->actor.params = 1;
+            this->actor.flags &= 0xFFFEFFFF;
```
```suggestion
            this->actor.flags &= ~0x10000;;
```

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/EnBji01_Update.asm")
+    if (limbIndex == 0xF) {
+        Math_Vec3f_Copy((Vec3f*)&sp20, &D_809CDCC8);
```
sp20 is already a Vec3f, so &sp20 should be fine for this function

rylieb replied:
Will be fixed in a batch commit.

kyleburnette commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.h:
```diff
@@ -2,12 +2,31 @@
+    /* 0x144 */ EnBji01ActionFunc actionFunc;
+    /* 0x148 */ SkelAnime skelAnime;
+    /* 0x18C */ ColliderCylinder collider;
+    /* 0x1D8 */ Vec3s jointTable[0x10]; 
```
We generally have the size of these in decimal

rylieb replied:
Will be fixed in a batch commit.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnBji01_Update(Actor* thisx, GlobalContext* globalCtx) {
+
+    static s16 sBlinkSequence[6] = { 0, 1, 2, 1, 0, 0 };
+    EnBji01* this = THIS;
+
+    s32 pad; /*pad until 4 main functions take gamestate that gets promoted to globalctx*/
```
The comment isn't needed here, having `pad` all over the place is pretty common.
```suggestion
void EnBji01_Update(Actor* thisx, GlobalContext* globalCtx) {
    static s16 sBlinkSequence[] = { 0, 1, 2, 1, 0, 0 };
    EnBji01* this = THIS;
    s32 pad;
```

kyleburnette reviewed (changes requested):
Great job on your first actor! Just a couple notes on things that jumped out at me at first glance. Also, it seems like your PR isn't building, but Jenkins is 404ing for me so I can't see why, nice.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+    EnBji01* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CD6B0.asm")
+    if ((limbIndex == 0) && ((globalCtx->gameplayFrames & 1) != 0)) {
```
```suggestion
    if ((limbIndex == 0) && ((globalCtx->gameplayFrames % 2) != 0)) {
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +47,421 @@ const ActorInit En_Bji_01_InitVars = {
+void EnBji01_Draw(Actor* thisx, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CDB04.asm")
+    static void* sEyeTextures[3] = { D_060049F0, D_06004E70, D_06005270 };
```
```suggestion
    static void* sEyeTextures[] = { D_060049F0, D_06004E70, D_06005270 };
```

AngheloAlf reviewed (commented):
Welcome to MM decomp!
You made a very good job for your first actor. Congrats
I left a few notes, but it looks good overall.
Also, Jenkins is mad about this PR. If you need any help fixing that feel free to ask.

EllipticEllipsis commented on src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.h:
```diff
@@ -10,7 +10,7 @@ typedef void (*BgHakaTombActionFunc)(struct BgHakaTomb*, GlobalContext*);
     /* 0x000 */ DynaPolyActor dyna;
     /* 0x15C */ BgHakaTombActionFunc actionFunc;
-    /* 0x160 */ s16 unk160[2];
+    /* 0x160 */ s16 cutscenes[1]; /* s16 alignment pad by compiler */
```
```suggestion
    /* 0x160 */ s16 cutscenes[1];
```

EllipticEllipsis commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +48,423 @@ const ActorInit En_Bji_01_InitVars = {
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CD6C0.asm")
+    switch (limbIndex) {
+        case 8: /* Torso */
+            rot->x += this->torsoXRotAdj;
+            rot->z += this->torsoZRotAdj;
+            break;
+        case 15: /* Head */
```
You can make an enum for the limbs if you want. There's a few examples in the repo, e.g. Romani.

rylieb replied:
This will be fixed in an upcoming batch commit.

hensldm commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +48,418 @@ const ActorInit En_Bji_01_InitVars = {
+            if (func_80147624(globalCtx) != 0) {
+                this->actor.flags &= ~0x10000;
+                switch (globalCtx->msgCtx.unk11F04) {
+                    case 1502:
```
This switch cases look to be textIds which are generally put as hex.

AngheloAlf replied:
Even if those aren't textIds, we usually left numbers in hex if we don't know what they are.

rylieb commented:
I've addressed all comments and suggestions up to this point. The only thing I couldn't fix without losing matching was the single early return. It could just be that's how the developers coded that one.

hensldm commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+                switch (globalCtx->msgCtx.unk11F04) {
+                    case 0x5DE:
+                        func_8013E1C8(&this->skelAnime, D_809CDC7C, 3, &this->animationIndex);
+                        func_80151938(globalCtx, 0x5DFU);
```
```suggestion
                        func_80151938(globalCtx, 0x5DF);
```

hensldm commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+                        func_80151938(globalCtx, 0x5DFU);
+                        break;
+                    case 0x5E4:
+                        func_80151938(globalCtx, 0x5E7U);
```
```suggestion
                        func_80151938(globalCtx, 0x5E7);
```

hensldm commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+                        func_80151938(globalCtx, 0x5E7U);
+                        break;
+                    case 0x5E5:
+                        func_80151938(globalCtx, 0x5E0U);
```
```suggestion
                        func_80151938(globalCtx, 0x5E0);
```

hensldm commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+                        func_80151938(globalCtx, 0x5E0U);
+                        break;
+                    case 0x5E7:
+                        func_80151938(globalCtx, 0x5E5U);
```
```suggestion
                        func_80151938(globalCtx, 0x5E5);
```

hensldm reviewed (commented):
Just a few more trailing U's. Good work on your first actor.

jpburnett commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.h:
```diff
@@ -2,12 +2,56 @@
+    /* 0x144 */ EnBji01ActionFunc actionFunc;
+    /* 0x148 */ SkelAnime skelAnime;
+    /* 0x18C */ ColliderCylinder collider;
+    /* 0x1D8 */ Vec3s jointTable[16]; 
+    /* 0x238 */ Vec3s morphTable[16];
```
```suggestion
    /* 0x1D8 */ Vec3s jointTable[BJI_LIMB_MAX]; 
    /* 0x238 */ Vec3s morphTable[BJI_LIMB_MAX];
```

jpburnett reviewed (commented):
Shikashi? more like Sick-kashi! The only minor detail I could find was using the ENUM vaule for the arrays in the header.

rylieb commented:
Hahaha thanks! Good suggestion, I just committed it.

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+    EnBji01* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 30.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600578C, &D_06000FDC, this->jointTable, this->morphTable, 0x10);
```
Now that you have a limb enum, you can use the LIMB_MAX here.
```suggestion
    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600578C, &D_06000FDC, this->jointTable, this->morphTable, BJI_LIMB_MAX);
```

jpburnett commented:
🌑 🧙

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+    s32 temp_f4 = 0;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CDA4C.asm")
+    if (limbIndex == 15) {
+        Math_Vec3f_Copy(&sp20, &D_809CDCC8);
```
Now that you have the limb enum, you can use it to remove those magic numbers. 
```suggestion
    if (limbIndex == BJI_LIMB_HEAD) {
        Math_Vec3f_Copy(&sp20, &D_809CDCC8);
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+    EnBji01* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/EnBji01_Destroy.asm")
+    if ((limbIndex == 0) && ((globalCtx->gameplayFrames % 2) != 0)) {
```
```suggestion
    if ((limbIndex == BJI_LIMB_NONE) && ((globalCtx->gameplayFrames % 2) != 0)) {
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+    if ((limbIndex == 0) && ((globalCtx->gameplayFrames % 2) != 0)) {
+        *dList = NULL;
+    }
+    if (limbIndex == 0) {
```
```suggestion
    if (limbIndex == BJI_LIMB_NONE) {
```

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+    this->actionFunc = func_809CD328;
+}
+
+void func_809CD328(EnBji01* this, GlobalContext* globalCtx) {
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        case 0:
```
Just as a note, function `func_80152498` is some kind of "GetDialogueState" (or whatever), in the sense the number it returns tells you the current state of the conversation (player answered a question, a message block ended, the whole conversation ended, etc).

We usually name this kind of function `_DialogueHandler`.

EnMa4 (https://github.com/zeldaret/mm/blob/9637e19303bdd2c99b1a03390cc1bbea786a8150/src/overlays/actors/ovl_En_Ma4/z_en_ma4.c#L649) has an example of this and a few notes on the values it uses. I though you may be interested on knowing this.

rylieb replied:
Awesome! I'll rename it when i get home.

rylieb replied:
Just ran rename_sym.sh and pushed it :)

AngheloAlf commented on src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c:
```diff
@@ -21,36 +44,405 @@ const ActorInit En_Bji_01_InitVars = {
+        case ENBJI01_PARAMS_FINISHED_CONVERSATION:
+            switch (gSaveContext.playerForm) {
+                case PLAYER_FORM_DEKU:
+                    if (gSaveContext.weekEventReg[17] & 0x10) {
+                        if (gSaveContext.weekEventReg[74] & 0x80) {
```
We don't really have decided if we should use dec or hex for weekEventReg's indexes, so either is fine.
I personally prefer to use hex because the most up-to-date sheet uses hex for its indexes.
That said, you can leave them whatever the way you prefer.

rylieb replied:
KZ uses hex which is why I converted the spreadsheet to it, but it used to be decimal, and i was under the impression thatdecomp used decimal as well, so I did decimal and I added a decimal column to the spreadsheet, so that either decomp or kz people can interperet it.

AngheloAlf reviewed (commented):
A few more stuff I missed last night.

rylieb commented:
![En_Bji_01 gv](https://user-images.githubusercontent.com/31797144/126578517-d0758d5d-fa41-4e94-810a-00a71368a640.png)
New graphovl after the rename

rylieb commented:
func_809CD634 (the one between dialogue handler and do nothing) is almost certainly responsible for loading a new scene (TF telescope in particular), but when I was doing my documentation pass, it was suggested I not rename it since there are two /code functions calls at the top that aren't in Warp_Uzu (the PF telescope) and it's unclear what they do.

rylieb commented:
Also all comments/suggestions as of this point have been addressed and resolved. Thanks everyone for your helpful suggestions so far :)

Kenix3 reviewed (approved):
Welcome to MM decomp! PR is looking good.
